targetselector/running_target_selector.go

```diff
@@ -144,6 +153,15 @@ func (r runningTargetSelector) GetMatchingNodesOverTotalNodes(c client.Client, i
 			}
 		}
 
+		if instance.Spec.Filters != nil && len(instance.Spec.Filters.Annotations) > 0 {
+			for k, v := range instance.Spec.Filters.Annotations {
+				nodeAnno, ok := node.Annotations[k]
+				if !ok || nodeAnno != v {
+					continue
+				}
+			}
+		}
+
```

Comment:
I don't like this duplicated code but I can't figure out a nice way to handle this for both kind of resources (pods and nodes) in a single function.

Reply:
I think it's okay duplication! It's probably more extra LoC to write a block of code that works for both